### PR TITLE
chore(TS): Swap an any for the actual type in toCanvasElement and toDataUrl

### PR DESCRIPTION
**Pull request by asturur**
## Description

When @jiayihu made the toCanvasElement option i noticed that we had an any to describe the types.
Today i was using the function and so i typed it.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "packages": {
     "": {
       "name": "fabric",
-      "version": "6.0.0-beta18",
+      "version": "6.0.0-beta19",
```
Comment (ShaMan123) [inline package-lock.json:9]: we should run `npm i` when we release and commit this as part of the action

File: src/shapes/Object/Object.ts [hunk 2]
```diff
@@ -70,6 +71,34 @@ export type TCachedFabricObject<T extends FabricObject = FabricObject> = T &
...
+  enableRetinaScaling?: boolean;
+  // Remove current object transform ( no scale , no angle, no flip, no skew ). Introduced in 2.3.4
+  withoutTransform?: boolean;
+  // Remove current object shadow. Introduced in 2.4.2
+  withoutShadow?: boolean;
+  // Account for canvas viewport transform
+  viewportTransform?: boolean;
+  // Function to create the output canvas to export ont
+  canvasProvider?: (el?: HTMLCanvasElement) => Canvas;
+}
```
Comment (ShaMan123) [inline src/shapes/Object/Object.ts:96]: I think if you use `/** */` as comment syntax then it appears when you hover the option in VSCODE

File: src/shapes/Object/Object.ts [hunk 3]
```diff
@@ -70,6 +71,34 @@ export type TCachedFabricObject<T extends FabricObject = FabricObject> = T &
...
+  // Enable retina scaling for clone image. Introduce in 1.6.4
+  enableRetinaScaling?: boolean;
+  // Remove current object transform ( no scale , no angle, no flip, no skew ). Introduced in 2.3.4
+  withoutTransform?: boolean;
+  // Remove current object shadow. Introduced in 2.4.2
+  withoutShadow?: boolean;
+  // Account for canvas viewport transform
+  viewportTransform?: boolean;
+  // Function to create the output canvas to export ont
+  canvasProvider?: (el?: HTMLCanvasElement) => Canvas;
```
Comment (ShaMan123) [inline src/shapes/Object/Object.ts:95]:
```suggestion
  canvasProvider?: (el?: HTMLCanvasElement) => StaticCanvas;
```

Comment (ShaMan123) [reply src/shapes/Object/Object.ts:95]: Or maybe we should pass a generic here `<T extends StaticCanvas>`

Comment (asturur) [reply src/shapes/Object/Object.ts:95]: yes maybe a generic is better. i ll see if i can make it without triggerin 1000 warning. If i trigger 1000 warning i l leave it as StaticCanvas

Comment (ShaMan123) [reply src/shapes/Object/Object.ts:95]: It might because of a def cycle

Comment (ShaMan123) [reply src/shapes/Object/Object.ts:95]: Maybe better to define the interface needed for the method

**Pull request merged**